esp-hal/src/dma/pdma/crypto.rs

```diff
@@ -0,0 +1,525 @@
+use esp32s2::crypto_dma::aes_sha_select::SELECT;
+use portable_atomic::{AtomicBool, Ordering};
+
+use crate::{
+    asynch::AtomicWaker,
+    dma::*,
+    interrupt::Priority,
+    peripheral::Peripheral,
+    peripherals::Interrupt,
+};
+
+pub(super) type CryptoRegisterBlock = crate::peripherals::crypto_dma::RegisterBlock;
+
+/// The RX half of a Crypto DMA channel.
+pub struct CryptoDmaRxChannel(pub(crate) CryptoDmaChannel);
+
+impl crate::private::Sealed for CryptoDmaRxChannel {}
+impl DmaRxChannel for CryptoDmaRxChannel {}
+impl Peripheral for CryptoDmaRxChannel {
+    type P = Self;
+
+    unsafe fn clone_unchecked(&self) -> Self::P {
+        Self(self.0.clone_unchecked())
+    }
+}
+
+/// The TX half of a Crypto DMA channel.
+pub struct CryptoDmaTxChannel(pub(crate) CryptoDmaChannel);
+
+impl crate::private::Sealed for CryptoDmaTxChannel {}
+impl DmaTxChannel for CryptoDmaTxChannel {}
+impl Peripheral for CryptoDmaTxChannel {
+    type P = Self;
+
+    unsafe fn clone_unchecked(&self) -> Self::P {
+        Self(self.0.clone_unchecked())
+    }
+}
+
+impl RegisterAccess for CryptoDmaTxChannel {
+    fn reset(&self) {
+        let register_block = self.0.register_block();
+        register_block.conf().modify(|_, w| {
+            w.out_rst().set_bit();
+            w.ahbm_rst().set_bit();
+            w.ahbm_fifo_rst().set_bit()
+        });
+        register_block.conf().modify(|_, w| {
+            w.out_rst().clear_bit();
+            w.ahbm_rst().clear_bit();
+            w.ahbm_fifo_rst().clear_bit()
+        });
+    }
```

Comment:
This isn't quite right. This should only toggle the `out_rst` bit.

Reply:
If you don't mind, I'll just keep doing what the TRM says until we actually run into an issue with it.

Reply:
Ehh fair enough, it'd be inconsistent with the other PDMA channels but it'll probably do the same thing.

Reply:
Looks like the other impls don't follow the TRM, then. FWIW they might not have been written with the S2's TRM in mind and ESP32 is a little more vague (`Reset the DMA state machine and FIFO parameters;`).

Reply:
Yeah, after this PR we can decide on the semantics of `reset()`, then make it consistent across everywhere.